src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -146,6 +112,66 @@ bool same_host_mem(cldnn::memory::ptr memPtr, uint8_t* hostPtr) {
 namespace ov {
 namespace intel_gpu {
 
+template<typename src_dt, typename dst_dt>
+void InferRequest::copyResultToOutputBlob(cldnn::memory::ptr src, InferenceEngine::Blob::Ptr dst, bool need_convert) {
+    auto& stream = m_graph->GetNetwork()->get_stream();
+
+    auto layout = src->get_layout();
+    auto size = layout.get_tensor();
+
+    auto locked_dst = dst->buffer();
+    auto dst_ptr = locked_dst.as<dst_dt*>();
+    OPENVINO_ASSERT(dst_ptr, "[GPU] Invalid output blob");
+
+    InferenceEngine::Blob::Ptr intermediate_blob = nullptr;
+
+    // Use copy_to() directly if: (1) convert not needed, and (2) data not padded
+    if (!need_convert && !layout.data_padding) {
+        src->copy_to(stream, dst_ptr);
+    } else {
+        auto desc = dst->getTensorDesc();
+        desc.setPrecision(PrecisionFromDataType(layout.data_type));
+
+        Blob::Ptr intermediate_blob = create_host_blob(desc, layout.is_dynamic());
```

Comment:
Also I think this blob can be reused between iterations to minimize overhead on realloc.